DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,63 @@
+  - [Developer Guide](#developer-guide)
+  - [Prerequisites](#prerequisites)
+  - [Install Docker Image](#install-docker-image)
+  - [Running Tests](#running-tests)
+  - [Client Code Generator](#client-code-generator)
+
+# Developer Guide
+
+## Prerequisites
+
+dotnet 6.0 sdk is required
+
+```
+$ dotnet --list-sdks
+6.0.xxx
+```
+
+
+
+## Install Docker Image
+
+Integration tests require [docker](https://opensearch.org/docs/latest/install-and-configure/install-opensearch/docker/).
+
+Run the following commands to install the docker image:
+
+```
+docker pull opensearchproject/opensearch:latest
+```
+
+Integration tests will auto-start the docker image. To start it manually:
+
+```
+docker run -d -p 9200:9200 -p 9600:9600 -e "discovery.type=single-node" opensearchproject/opensearch:latest
+```

```

Comment:
The client's integration tests don't require you to have a docker image running, it will download and setup temporary OpenSearch clusters itself.

Reply:
I might be misunderstanding but the guide does say the integration tests will auto-start the docker image, do you want it worded differently or?

Happy to remove the part about pulling the image and running it manually though.

Reply:
The .NET integrations tests don't use docker at all, so we don't need to mention docker. The integration tests download and cache the OpenSearch release tarball then use it to automatically spin up clusters as necessary for testing.